manifest.go

```diff
@@ -1,76 +1,175 @@
+	ProviderNameSpace          string       `xml:"ProviderNameSpace"`
+	Type                       string       `xml:"Type"`
+	Version                    string       `xml:"Version"`
+	Label                      string       `xml:"Label"`
+	HostingResources           string       `xml:"HostingResources"`
+	MediaLink                  string       `xml:"MediaLink"`
+	Certificate                *certificate `xml:"Certificate,omitempty"`
+	PublicConfigurationSchema  string       `xml:"PublicConfigurationSchema,omitempty"`
+	PrivateConfigurationSchema string       `xml:"PrivateConfigurationSchema,omitempty"`
+	Description                string       `xml:"Description"`
+	BlockRoleUponFailure       string       `xml:"BlockRoleUponFailure,omitempty"`
+	IsInternalExtension        bool         `xml:"IsInternalExtension"`
+	Eula                       string       `xml:"Eula,omitempty"`
+	PrivacyURI                 string       `xml:"PrivacyUri,omitempty"`
+	HomepageURI                string       `xml:"HomepageUri,omitempty"`
+	IsJSONExtension            bool         `xml:"IsJsonExtension,omitempty"`
+	CompanyName                string       `xml:"CompanyName,omitempty"`
+	SupportedOS                string       `xml:"SupportedOS,omitempty"`
+	Regions                    string       `xml:"Regions,omitempty"`
+}
+
+type extensionImageGlobal struct {
+	XMLName                    string       `xml:"ExtensionImage"`
+	NS                         string       `xml:"xmlns,attr"`
+	ProviderNameSpace          string       `xml:"ProviderNameSpace"`
+	Type                       string       `xml:"Type"`
+	Version                    string       `xml:"Version"`
+	Label                      string       `xml:"Label"`
+	HostingResources           string       `xml:"HostingResources"`
+	MediaLink                  string       `xml:"MediaLink"`
+	Certificate                *certificate `xml:"Certificate,omitempty"`
+	PublicConfigurationSchema  string       `xml:"PublicConfigurationSchema,omitempty"`
+	PrivateConfigurationSchema string       `xml:"PrivateConfigurationSchema,omitempty"`
+	Description                string       `xml:"Description"`
+	BlockRoleUponFailure       string       `xml:"BlockRoleUponFailure,omitempty"`
+	IsInternalExtension        bool         `xml:"IsInternalExtension"`
+	Eula                       string       `xml:"Eula,omitempty"`
+	PrivacyURI                 string       `xml:"PrivacyUri,omitempty"`
+	HomepageURI                string       `xml:"HomepageUri,omitempty"`
+	IsJSONExtension            bool         `xml:"IsJsonExtension,omitempty"`
+	CompanyName                string       `xml:"CompanyName,omitempty"`
+	SupportedOS                string       `xml:"SupportedOS,omitempty"`
+	Regions                    string       `xml:"Regions"`
+}
+
+type extensionManifest interface {
+	Marshal() ([]byte, error)
+}
+
+func isGuestAgent(providerNameSpace string) bool {
+	return "Microsoft.OSTCLinuxAgent" == providerNameSpace
```

Comment:
It is a different deployment mechanism, and I don't think it is going to change. ☹️